Examples/Examples/FeatureFormExampleView.swift

```diff
@@ -147,70 +168,204 @@ private extension URL {
+        case .applyingEdits:
+            Text("Applying edits")
+        default:
+            Text("")
+        }
+    }
+    
+    // MARK: Methods
     
     /// Reverts any local edits that haven't yet been saved to service geodatabase.
     func discardEdits() {
-        featureForm?.discardEdits()
-        featureForm = nil
+        guard case let .cancellationPending(featureForm) = state else {
+            return
+        }
+        featureForm.discardEdits()
+        state = .idle
     }
     
     /// Submit the changes made to the form.
     func submitChanges() async {
-        guard let featureForm,
-              let table = featureForm.feature.table as? ServiceFeatureTable,
-              let database = table.serviceGeodatabase else {
-            print("A precondition to submit the changes wasn't met.")
+        guard case let .editing(featureForm) = state else { return }
+        await validateChanges(featureForm)
+    }
+    
+    // MARK: Private methods
+    
+    /// Applies edits to the remote service.
+    private func applyEdits(_ featureForm: FeatureForm, _ table: ServiceFeatureTable) async {
+        state = .applyingEdits(featureForm)
+        guard let database = table.serviceGeodatabase else {
+            state = .generalError(featureForm, Text("No geodatabase found."))
             return
         }
-        
-        guard table.isEditable else {
-            submissionError = Text("The feature table isn't editable.")
+        guard database.hasLocalEdits else {
+            state = .generalError(featureForm, Text("No database edits found."))
             return
         }
-        
-        guard featureForm.validationErrors.isEmpty else {
-            submissionError = Text("The form has ^[\(featureForm.validationErrors.count) validation error](inflect: true).")
+        let resultErrors: [Error]
+        do {
+            if let serviceInfo = database.serviceInfo, serviceInfo.canUseServiceGeodatabaseApplyEdits {
+                let featureTableEditResults = try await database.applyEdits()
+                resultErrors = featureTableEditResults.flatMap { featureTableEditResult in
+                    checkFeatureEditResults(featureForm, featureTableEditResult.editResults)
+                }
+            } else {
+                let featureEditResults = try await table.applyEdits()
+                resultErrors = checkFeatureEditResults(featureForm, featureEditResults)
+            }
+        } catch {
+            state = .generalError(featureForm, Text("The changes could not be applied to the database or table.\n\n\(error.localizedDescription)"))
             return
         }
-        
-        try? await table.update(featureForm.feature)
-        
-        guard database.hasLocalEdits else {
-            print("No submittable changes found.")
+        if resultErrors.isEmpty {
+            state = .idle
+        } else {
+            state = .generalError(featureForm, Text("Changes were not applied."))
```

Comment:
Maybe a comment here stating the client could display the errors to the user, if so desired.
```suggestion
            // Additionally, you could display the errors to the user using `resultErrors`.
            state = .generalError(featureForm, Text("Changes were not applied."))
```